### ISSUE TITLE
Improve error message for "Could not resolve" when using `bun build`

### ISSUE THREAD

**Issue by vinliao** [oven-sh/bun#6131]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.4.15-100.fc37.x86_64 x86_64 x86_64

### What steps can reproduce the bug?

```
// 1.ts
import { sleep } from "bun";

await sleep(5000);
console.log("yay");
```

Build the file by running 
bun build ./1.ts --outfile=1.js

### What is the expected behavior?

It should build without fail

### What do you see instead?

```
error: Could not resolve: "bun". Maybe you need to "bun install"?
import { sleep } from "bun";
                      ^
```

**Comment by cyfung1031** [comment]
You don't need to import anything. and Bun is a class so "Bun" not "bun"

```js
await Bun.sleep(5000);
console.log("yay");
```

**Comment by SoundOfTheSky** [comment]
By default bun targets browsers. Use `bun build --target=bun`

**Comment by Electroid** [comment]
> By default bun targets browsers. Use bun build --target=bun

This is correct. We should change the error message to make this more obvious.

**Comment by loudar** [comment]
Would still be cool to have a clearer error message ;)
